lib/pure/asyncmacro.nim

```diff
@@ -447,8 +344,9 @@ macro multisync*(prc: untyped): untyped =
   result = newStmtList()
   result.add(asyncSingleProc(asyncPrc))
   result.add(sync)
+  # echo result.repr
 
-proc await*[T](x: T) =
-  ## The 'await' keyword is also defined here for technical
-  ## reasons. (Generic symbol lookup prepass.)
-  {.error: "Await only available within .async".}
+# overload for await as a fallback handler, based on the yglukhov's patch to chronos: https://github.com/status-im/nim-chronos/pull/47
+# template await*(f: typed): untyped =
+  # static:
+    # error "await only available within {.async.}"
```

Comment:
You still need this, no?

Reply:
oh, you removed this because you didn't want `dirty`. Is `dirty` really that bad? :)

Reply:
no, but it can be a bit surprising: we need to somehow hint here in the error "if you use a template, you might want to use `{.dirty.}`" because otherwise many users would be just confused by why just a template/macro can't work 

on the other hand just a type mismatch is very natural, `await` is not found and that's it